### PR TITLE
Passing IFluideCodeDetails in GetAbsoluteUrl to be able to get the package name from there.

### PR DESCRIPTION
**Pull request by abrahamrq**
Related to Issue #3947 
Related to PR #3944 

Moving FluidPackage.ts from container-definitions to core-interfaces
Passing the CodeDetails to GetAbsoluteUrl method in IUrlResolver so we are able to grab the package name from there

### CODE REVIEW

File: BREAKING.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ No back-compat or mix version impact.
 DriverHeader is a driver concept, so move from core-interface to driver-definitions. CreateNewHeader is also a kind of driver header, merged it into DriverHeader.
 
 ### IFluidPackage Changes
+- Moving IFluidPackage from "@fluidframework/container-definitions" to '@fluidframework/core-interfaces'
```
Comment (anthony-murphy) [inline BREAKING.md:17]:
```suggestion
- Moving IFluidPackage and IFluidCodeDetails from "@fluidframework/container-definitions" to '@fluidframework/core-interfaces'
```

File: packages/loader/core-interfaces/src/fluidPackage.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@
  /**
   * Specifies an environment on Fluid property of a IFluidPackage
   */
-export interface IFluidPackageEnvironment {
+ export interface IFluidPackageEnvironment {
```
Comment (anthony-murphy) [inline packages/loader/core-interfaces/src/fluidPackage.ts:9]:
```suggestion
export interface IFluidPackageEnvironment {
```

**Review by anthony-murphy — APPROVED** [review]
looks great. thanks

File: server/gateway/src/alfred.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@
  * Licensed under the MIT License.
  */
 
-import { IFluidCodeDetails } from "@fluidframework/container-definitions";
```
Comment (anthony-murphy) [inline server/gateway/src/alfred.ts:6]: nothing under server is part of the client mono-repo, we version bump this after release, so these should all be reverted for now

**Review by christiango — APPROVED** [review]

**Review by anthony-murphy — APPROVED** [review]

**Pull request merged**